muspinsim/simconfig.py

```diff
@@ -180,26 +182,55 @@ def __init__(self, params={}):
         if finfo["fit"]:
             if len(self._file_ranges) > 0:
                 raise MuSpinConfigError("Can not have file ranges when fitting")
-            logging.warning(
-                "'x_axis' values will be overwritten by the experimental data"
-            )
-            # The x axis is overridden, whatever it is
+
             xname = list(self._x_range.keys())[0]
             self._constants.pop(xname, None)  # Just in case it was here
 
-            # Special cases where loaded data will be a single value, but we
-            # actually need to load multiple
-            if xname in ["B", "intrinsic_B"]:
-                # Default to z axis (the same as in _validate_B below -
-                # the orientation is applied later)
-                self._x_range[xname] = np.array(
-                    [[0, 0, v] for v in finfo["data"][:, 0]]
-                )
-            else:
-                self._x_range[xname] = finfo["data"][:, 0]
-            if xname == "t":
-                # Special case
-                self._time_N = len(self._x_range[xname])
+            if self._x_range[xname] is None:
```

Comment:
The behaviour's a bit counter-intuitive here I think. We initialize to have a value of `None` in the `dict`:
https://github.com/muon-spectroscopy-computational-project/muspinsim/blob/08a343039aeb6e057e0820d6efcc059693ea1c66/muspinsim/simconfig.py#L117
Which may get assigned an actual range (or not).

Then `xname` is taken directly from the `dict`, so must be present as a key:
https://github.com/muon-spectroscopy-computational-project/muspinsim/blob/08a343039aeb6e057e0820d6efcc059693ea1c66/muspinsim/simconfig.py#L186

I'm naturally wary of changing things which aren't needed for the issue, to keep scope limited and for fear of breaking something accidentally. So while I think a lot of this could be rewritten to be clearer, at the time I left the lines I didn't need to change as is (e.g 186). But that means the new line 189 is confusing.

To make what's actually happening clearer, propose changing `keys()` to `items()` on 186 and assigning/using the value for the first key/value pair directly.